tekton/utils/pipeline_run_builder.go

```diff
@@ -215,28 +217,48 @@ func (b *PipelineRunBuilder) WithParamsFromConfigMap(configMap *corev1.ConfigMap
 func (b *PipelineRunBuilder) WithPipelineRef(pipelineRef *tektonv1.PipelineRef) *PipelineRunBuilder {
 	b.pipelineRun.Spec.PipelineRef = pipelineRef
 
-	if pipelineRef.Resolver == "git" {
-		for _, param := range pipelineRef.Params {
-			if param.Name == "revision" {
-				b.WithParams(tektonv1.Param{
-					Name: "taskGitRevision",
-					Value: tektonv1.ParamValue{
-						Type:      tektonv1.ParamTypeString,
-						StringVal: param.Value.StringVal,
-					},
-				})
+	if pipelineRef != nil && pipelineRef.ResolverRef.Resolver == "git" {
+		var gitURL, revision string
+
+		for _, param := range pipelineRef.ResolverRef.Params {
+			switch param.Name {
+			case "url":
+				gitURL = param.Value.StringVal
+			case "revision":
+				revision = param.Value.StringVal
 			}
+		}
 
-			if param.Name == "url" {
-				b.WithParams(tektonv1.Param{
-					Name: "taskGitUrl",
-					Value: tektonv1.ParamValue{
-						Type:      tektonv1.ParamTypeString,
-						StringVal: param.Value.StringVal,
-					},
-				})
+		var resolvedSHA string
+		var err error
+
+		resolvedSHA, err = git.ResolveBranchToSHA(gitURL, revision)
```

Comment:
No need to declare the variables upfront, isn't it?
```suggestion
		resolvedSHA, err := git.ResolveBranchToSHA(gitURL, revision)
```

Reply:
I wasn't aware about that good practice. I'll make sure I read about it! Thanks for pointing.